crates/solvers/src/domain/baseline.rs

```diff
@@ -0,0 +1,91 @@
+//! "Baseline" solver implementation.
+
+use crate::{
+    boundary,
+    domain::{auction, eth, liquidity, order, solution},
+};
+use std::{collections::HashSet, num::NonZeroUsize};
+
+use super::solution::Interaction;
+
+pub struct Baseline {
+    pub weth: eth::WethAddress,
+    /// Set of base-tokens for which to always consider when path-finding. This
+    /// allows paths of the kind `TOKEN1 -> WETH -> TOKEN2` to be considered.
+    pub base_tokens: HashSet<eth::TokenAddress>,
+    /// Maximum number of hops that can be considered in a trading path. A value
+    /// of 1 indicates that only a direct trade is allowed.
+    pub max_hops: NonZeroUsize,
+}
+
+impl Baseline {
+    /// Solves the specified auction, returning a vector of all possible
+    /// solutions.
+    pub fn solve(&self, auction: &auction::Auction) -> Vec<solution::Solution> {
+        let boundary_solver =
+            boundary::baseline::Solver::new(&self.weth, &self.base_tokens, &auction.liquidity);
+
+        auction
+            .orders
+            .iter()
+            .filter_map(|order| {
+                let route = boundary_solver.route(order::UserOrder::new(order)?, self.max_hops)?;
+
+                Some(solution::Solution {
+                    prices: solution::clearingprices! {
+                        order.sell.token => route.output().amount,
+                        order.buy.token => route.input().amount,
+                    },
+                    trades: vec![solution::Trade::fill(order.clone())],
+                    interactions: route
+                        .segments
+                        .iter()
+                        .map(|segment| Interaction {
+                            liquidity: segment.liquidity.clone(),
```

Comment:
The difference is I'm only cloning liquidity for the winning route (which is small) instead of all liquidity, which may be very large.

Is there a reason we shouldn't use references for segments?